sdk/core/Azure.Core/src/Response.cs

```diff
@@ -41,6 +41,42 @@ public abstract class Response : IDisposable
         /// </summary>
         public virtual ResponseHeaders Headers => new ResponseHeaders(this);
 
+        private BinaryData? _content;
+
+        /// <summary>
+        /// Gets the contents of HTTP response. Returns a BinaryData with no payload for responses without content.
+        /// </summary>
+        public virtual BinaryData Content
+        {
+            get
+            {
+                if (_content != null)
+                {
+                    return _content;
+                }
+
+                if (ContentStream == null)
+                {
+                    _content = new BinaryData(Array.Empty<byte>());
+                    return _content;
+                }
+
+                // If the underlying stream is a memory stream, we can just share the underlying buffer
+                // with the `BinaryData` we are going to return. Otherwise, we have to read the entire
+                // response into memory.
+                MemoryStream? memoryContent = ContentStream as MemoryStream;
+
+                if (memoryContent == null)
+                {
+                    memoryContent = new MemoryStream();
+                    ContentStream.CopyTo(memoryContent);
```

Comment:
Should we be trying to do anything more here?  If the user already read a few bytes from the Stream, we wouldn't end up copying everything, but we couldn't really seek to the start of the stream in all cases (we could do this in the case where CanSeek returned `true`, I guess).